stormtrooper/generative.py

```diff
@@ -33,12 +33,14 @@ class GenerativeZeroShotClassifier(BaseEstimator, ClassifierMixin):
 
     Parameters
     ----------
-    model_name: str, default 'stabilityai/StableBeluga-7B'
+    model_name: str, default 'HuggingFaceH4/zephyr-7b-beta'
```

Comment:
It seems odd to have both a generative interface and a openai interface. I would probably just have the generative interface and supply a "TextGeneratationPipeline", which follow a specific format. That way you can avoid having two very similar objects.

This is a bit of rewrite though so I am not sure it is worth it, but it would allow you to test both workflows without calling APIs etc.

this would also allow you to easily wrap new models.

Reply:
Also, good idea. For a release `1.0.0` I think something like this would be a great target.
Initially this would not have been feasible as: a) the API's of HuggingFace and OpenAI were significantly different b) I wanted to keep library code as simple, and decoupled as humanly possible.
I think for this release I will keep breaking changes to a minimium